src/core/Akka.Tests.Performance/Actor/ActorSelectionSpecs.cs

```diff
@@ -0,0 +1,125 @@
+            private readonly ManualResetEventSlim _resetEvent;
+
+            public BenchmarkActor(Counter counter, long maxExpectedMessages, ManualResetEventSlim resetEvent)
+            {
+                _counter = counter;
+                _maxExpectedMessages = maxExpectedMessages;
+                _resetEvent = resetEvent;
+            }
+
+            protected override void OnReceive(object message)
+            {
+                _counter.Increment();
+                if (++_currentMessages == _maxExpectedMessages)
+                    _resetEvent.Set();
+            }
+        }
+
+        [PerfSetup]
+        public void Setup(BenchmarkContext context)
+        {
+            _selectionOpCounter = context.GetCounter(ActorSelectionCounterName);
+            System = ActorSystem.Create("MailboxThroughputSpecBase" + Counter.GetAndIncrement());
+            _receiver = System.ActorOf(Props.Create(() => new BenchmarkActor(_selectionOpCounter, NumberOfMessages, _resetEvent)));
+            _receiverActorPath = _receiver.Path;
+            _oneMessageBenchmarkProps = Props.Create(() => new BenchmarkActor(_selectionOpCounter, 1, _resetEvent));
+        }
+
+        [PerfBenchmark(Description = "Tests the message delivery throughput of NEW ActorSelections to PRE-EXISTING actors", 
+            NumberOfIterations = 13, RunMode = RunMode.Throughput, RunTimeMilliseconds = 1000, TestMode = TestMode.Measurement)]
+        [CounterMeasurement(ActorSelectionCounterName)]
+        public void NewActorSelectionOnNewActorThroughput(BenchmarkContext context)
+        {
+            var actorRef = System.ActorOf(_oneMessageBenchmarkProps); // create a new actor every time
+            System.ActorSelection(actorRef.Path).Tell("foo"); // send that actor a message via selection
+            _resetEvent.Wait();
+            _resetEvent.Reset();
+        }
+
+        [PerfBenchmark(Description = "Tests the message delivery throughput of REUSABLE ActorSelections to PRE-EXISTING actors",
+            NumberOfIterations = 13, RunMode = RunMode.Iterations, TestMode = TestMode.Measurement)]
+        [CounterMeasurement(ActorSelectionCounterName)]
+        public void ReusedActorSelectionOnPreExistingActorThroughput(BenchmarkContext context)
+        {
+            var actorSelection = System.ActorSelection(_receiverActorPath);
+            for (var i = 0; i < NumberOfMessages;)
+            {
+                actorSelection.Tell("foo");
+                ++i;
+            }
+            _resetEvent.Wait();
+        }
+
+        [PerfBenchmark(Description = "Tests the message delivery throughput of NEW ActorSelections to NEW actors. This is really a stress test.",
+            NumberOfIterations = 13, RunMode = RunMode.Iterations, TestMode = TestMode.Measurement)]
+        [CounterMeasurement(ActorSelectionCounterName)]
+        public void NewActorSelectionOnPreExistingActorThroughput(BenchmarkContext context)
+        {
```

Comment:
How does this test differ from ReusedActorSelectionOnPreExistingActorThroughput ?
They both contain the same line `var actorSelection = System.ActorSelection(_receiverActorPath);`